checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -133,166 +411,116 @@ protected boolean checkOverride(MethodTree overriderTree,
 
-                            // A GuardedBy annotation on the receiver of the enclosing method was found.
-                            // Now check if the receiver on the enclosing method is the same as the receiver
-                            // on the field/method we are checking preconditions for.
+            Element invokedElement = TreeUtils.elementFromUse(expr);
 
-                            if (guardedByValue != null) {
-                                Node nodeNode = atypeFactory.getNodeForTree(tree);
-                                Node receiverNode = null;
+            AnnotatedTypeMirror atmOfReceiverOrIdentifier = atypeFactory.getReceiverType(tree);
 
-                                if (nodeNode instanceof FieldAccessNode) {
-                                    receiverNode = ((FieldAccessNode) nodeNode).getReceiver();
-                                }
-                                else if (nodeNode instanceof MethodInvocationNode)
-                                {
-                                    MethodAccessNode man = ((MethodInvocationNode) nodeNode).getTarget();
-
-                                    if (!man.getMethod().getModifiers().contains(Modifier.STATIC)) {
-                                        receiverNode = man.getReceiver();
-                                    }
-                                }
+            Node node = atypeFactory.getNodeForTree(tree);
 
-                                if (receiverNode instanceof ExplicitThisLiteralNode ||
-                                    receiverNode instanceof ImplicitThisLiteralNode) {
+            // Is the receiver of the expression being accessed the same as the receiver of the enclosing method?
+            boolean receiverIsThatOfEnclosingMethod = false;
 
-                                    // The receivers match. Add to the preconditions set.
-                                    for (String lockExpression : guardedByValue) {
+            Node receiverNode = null;
 
-                                        if (lockExpression.equals("itself")) {
-                                            // This is critical. That's because right now we know that, since
-                                            // we are dealing with the receiver of the method, "itself" corresponds
-                                            // to "this". However once super.checkPreconditions is called, that
-                                            // knowledge is lost and it will think that "itself" is referring to
-                                            // the variable the precondition we are about to add is attached to.
-                                            lockExpression = "this";
-                                        }
+            if (node instanceof FieldAccessNode || node instanceof MethodAccessNode) {
+                receiverNode = node instanceof FieldAccessNode ? ((FieldAccessNode) node).getReceiver() : ((MethodAccessNode) node).getReceiver();
```

Comment:
receiverNode is only used inside this if statement, so it should be declared here.